lib/findtoken.h

```diff
@@ -74,6 +74,10 @@ T* findToken(T* start, const Token* end, const Predicate& pred)
     return result;
 }
 
+bool findTokensSkipDeadCodeImplMatch_1(const Token* tok);
```

Comment:
I think it's unfortunate to separate the match patterns from the code. It can lead to bugs. Is it possible to move the whole function implementation to the cpp file..

Reply:
> I think it's unfortunate to separate the match patterns from the code. It can lead to bugs.

Yeah - not my intention at all to get this code in. As I wrote I want to get help on how to get this in properly.

> Is it possible to move the whole function implementation to the cpp file..

Since templates are involved that is problematic. I failed in my attempts so far.

Reply:
hmm.. I haven't tried to take this the whole way but I have the idea we could do something like this. Move the `findTokensSkipDeadCodeImpl` template to the findtoken.cpp file.

In findtoken.h:
```
std::vector<Token*> findTokensSkipDeadCodeCpp(const Library& library,
                                       Token* start,
                                       const Token* end,
                                       const std::function<bool(const Token*)>& pred,
                                       const std::function<std::vector<MathLib::bigint>(const Token*)>& evaluate);

std::vector<const Token*> findTokensSkipDeadCodeCpp(const Library& library,
                                       const Token* start,
                                       const Token* end,
                                       const std::function<bool(const Token*)>& pred,
                                       const std::function<std::vector<MathLib::bigint>(const Token*)>& evaluate);

template<class T, class Predicate, class Evaluate, REQUIRES("T must be a Token class", std::is_convertible<T*, const Token*> )>
std::vector<T*> findTokensSkipDeadCode(const Library& library,
                                       T* start,
                                       const Token* end,
                                       const Predicate& pred,
                                       const Evaluate& evaluate)
{
    std::function<bool(const Token*)> predFunc = pred;
    std::function<std::vector<MathLib::bigint>(const Token*)> evaluateFunc = evaluate;
    return findTokensSkipDeadCodeCpp(library, start, end, predFunc, evaluateFunc);
}
```

The `findTokensSkipDeadCodeCpp` are defined in the findtoken.cpp and invokes the `findTokensSkipDeadCodeImpl`.

The code does get more messy as there are "Cpp" intermediate functions..

Reply:
I have a solution which builds. It is a bit messier than the hack in terms of total changes. I will give it a spin and update the PR later.

Reply:
Done. It is even a tad faster than the previous hack. Will post profiling numbers later.